fvm/src/executor/validator.rs

```diff
@@ -0,0 +1,132 @@
+use anyhow::anyhow;
+use cid::Cid;
+use fvm_ipld_encoding::{Cbor, RawBytes, DAG_CBOR};
+use fvm_shared::error::ExitCode;
+use fvm_shared::message::params::ValidateParams;
+use fvm_shared::message::Message;
+
+use super::{ApplyKind, ApplyRet, DefaultExecutor, Executor, ValidateExecutor, ValidateRet};
+use crate::call_manager::{CallManager, ExecutionType, InvocationResult};
+use crate::kernel::{Block, Context, ExecutionError};
+use crate::machine::Machine;
+use crate::Kernel;
+
+/// TODO try not to be stuck with Default, but it has methods methods i want for validate.
+pub struct DefaultValidateExecutor<K: Kernel>(pub DefaultExecutor<K>);
```

Comment:
Yeah, let's just stick to a single type unless there's a strong reason to wrap this.